grammars/silver/compiler/extension/records/Records.sv.md

```diff
@@ -0,0 +1,370 @@
+# Records
+
+```silver imports
+grammar silver:compiler:extension:records;
+
+import silver:compiler:definition:core;
+import silver:compiler:definition:env;
+import silver:compiler:definition:type;
+import silver:compiler:definition:type:syntax;
+```
+
+The records extension is meant to codify a common pattern of defining product types that generates efficient code and has nice usability.
+
+## Declaration
+
+A record definition starts with the `record` keyword.
+
+```silver
+terminal RecordKwd_t 'record' lexer classes {KEYWORD};
+```
+
+A record definition looks basically like a nonterminal declaration followed by a list of fields.
+This forwards to a nonterminal, a single production with no children, and an annotation that occurs on the nonterminal for each field.
+
+```silver
+concrete production recordDecl
+this::AGDcl ::= quals::NTDeclQualifiers  'record'  name::IdUpper_t
```

Comment:
I would consider it a style guideline in the Silver compiler itself to use `top` for the LHS name, just to be consistent.

Reply:
Ah, I'll fix that.